### DISCUSSION TITLE
AWS SES email provider single server config fails when object config is working.

### DISCUSSION THREAD

**Discussion by lelabo-m** [nextauthjs/next-auth#5136] [Help]
### Provider type

Email

### Environment

 System:
    OS: Linux 5.10 Ubuntu 20.04.4 LTS (Focal Fossa)
    CPU: (8) x64 Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz
    Memory: 4.37 GB / 7.72 GB
    Container: Yes
    Shell: 5.0.17 - /bin/bash
  Binaries:
    Node: 16.13.0 - ~/.nvm/versions/node/v16.13.0/bin/node
    npm: 8.16.0 - ~/.nvm/versions/node/v16.13.0/bin/npm
  Browsers:
    Chrome: 104.0.5112.79
    Firefox: 103.0
  npmPackages:
    next: 12.2.3 => 12.2.3 
    next-auth: ^4.10.3 => 4.10.3 
    react: 18.2.0 => 18.2.0 

### Reproduction URL

NA

### Describe the issue

I tried to set up an email provider with AWS SES as the SMTP server but stumbled into this kind of strange behavior :
The single line config does not work but the object configuration with each parameter separated does work.

From the error I am getting in my terminal, I can see that the username was normalized to lowercase.
I am thinking this may be the issue because trying to use SES with a lowercase username does not work.

### How to reproduce

I was trying to configure an email provider with the following configuration: 
```
    EmailProvider({
      server: env.EMAIL_SERVER,
      from: env.EMAIL_LOGIN_FROM,
    }),
```
with the EMAIL_SERVER being something like : ` smtp://username:password@email-smtp.eu-west-2.amazonaws.com:587`

and I was getting the following error in my terminal:

  error: {
    message: 'getaddrinfo ENOTFOUND username',
    stack: 'Error: getaddrinfo ENOTFOUND username\n' +
      '    at GetAddrInfoReqWrap.onlookupall [as oncomplete] (node:dns:79:26)',
    name: 'Error'
  },
  providerId: 'email',
  message: 'getaddrinfo ENOTFOUND username'
}

### Expected behavior

Both configurations should work because I used the same parameters (double-checked it multiple times).
I suspect something between next-auth and nodemailer is normalizing the connection string.

**Comment by lelabo-m** [comment]
Ok, I think it is caused by the latest bugfix...
```
providers: add normalizeIdentifier to EmailProvider 
```
I don't know what was the reasoning behind this, but it breaks some configuration flows.

Maybe a section or a warning in the doc to explain the situation (sorry if it is already done).

**Comment by rlancejohnson** [comment]
I am facing this issue as well, only I get the error below whether I am using the single server config or the object config and with or without the normalizeIdentifier function, suggested in the [docs](https://next-auth.js.org/providers/email#normalizing-the-email-address), added to the EmailProvider. I am using the Prisma and MongoDB. The verification token is being created in the database correctly. There is just some issue with the email being sent from nodemailer from what I can tell.

I have completed the testing in this [AWS SES doc](https://docs.aws.amazon.com/ses/latest/dg/send-email-smtp-client-command-line.html) successfully, so I know that I have everything configured in AWS SES correctly and the right credentials.

Any ideas on what I could be missing? Any help would be very appreciated.

```
[next-auth][error][SIGNIN_EMAIL_ERROR] 
https://next-auth.js.org/errors#signin_email_error getaddrinfo ENOTFOUND ses_username {
  error: {
    message: 'getaddrinfo ENOTFOUND ses_username',
    stack: 'Error: getaddrinfo ENOTFOUND ses_username\n' +
      '    at GetAddrInfoReqWrap.onlookupall [as oncomplete] (node:dns:118:26)',
    name: 'Error'
  },
  providerId: 'email',
  message: 'getaddrinfo ENOTFOUND ses_username'
}
```

**Comment by seenasabti** [comment]
for anyone coming across this. this will resolve it:
```
import EmailProvider from "next-auth/providers/email";
...
providers: [
  EmailProvider({
    server: {
      host: process.env.EMAIL_SERVER_HOST,
      port: process.env.EMAIL_SERVER_PORT,
      auth: {
        user: process.env.EMAIL_SERVER_USER,
        pass: process.env.EMAIL_SERVER_PASSWORD
      }
    },
    from: process.env.EMAIL_FROM
  }),
],
```

**Reply by TeemuKoivisto** [reply]
Yeah I have no idea why this object config is needed instead of connection string but it works. Thanks!
